test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ESClientYamlSuiteTestCase.java

```diff
@@ -367,4 +373,76 @@ private String errorMessage(ExecutableSection executableSection, Throwable t) {
     protected boolean randomizeContentType() {
         return true;
     }
+
+        /**
+     * Sniff the cluster for host metadata if it hasn't already been sniffed. This isn't the
+     * same thing as using the {@link Sniffer} because:
+     * <ul>
+     * <li>It doesn't replace the hosts that that {@link #client} communicates with
+     * <li>If there is already host metadata it skips running. This behavior isn't
+     *     thread safe but it doesn't have to be for our tests.
+     * </ul>
+     */
+    private void sniffHostMetadata(RestClient client) throws IOException {
```

Comment:
How about removing the need for this and for boundHosts (I think!) by enabling sniffing in our yaml tests? wouldn't that be a good way to test it too?

Reply:
I *think* that'd be safe, but it feels like it is a larger change than this. I'll have to think about it some more.